### DISCUSSION TITLE
Vercel + Nuxt redirect subdomain to path

### DISCUSSION THREAD

**Discussion by mroddev** [vercel/community#2480] [Help] [answered]
### Problem Description

I followed this [article](https://vercel.com/guides/can-i-redirect-from-a-subdomain-to-a-subpath) to get a subdomain to redirect to a domain path but I can't seem to get this to work.

I want to redirect 'en.website.com' to 'website.com/en' and also all respective pages, for example: 'en.website.com/some-page' to 'website.com/en/some-page'.

By doing it with vercel.json nothing happens. With the 'en.website.com' domain added on vercel, when accessing it, it just displays the website frontpage, there is no redirect.
If I remove the 'en.website.com' domain from vercel, when accessing it, it just shows a vercel 404 page.

This is my current vercel.json file:
``` 
{ 
  "rewrites": [
    {
      "source": "/:path*",
      "has": [
        {
          "type": "host",
          "value": "en.website.com"
        }
      ],
      "destination": "/en/:path*"
    }
  ]
}

```

**Comment by amyegan** [comment, marked as answer]
I want to make sure we're talking about the same thing. [Redirects](https://vercel.com/docs/concepts/projects/project-configuration#redirects) send users to a new URL which is displayed as the page address. [Rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites) display content from a different URL but do not change the original address that was accessed.

I don't know your site setup, so please forgive me if these questions seem silly. Should the content displayed at `website.com/en` be different from what is being shown at `en.website.com`? Do you get the result you expected if you change `rewrites` to `redirects`?

I know that sometimes rewriting the root path can get a bit of tricky. Does it work if you add a rewrite for the root path specifically? Then the vercel.json file would look like...

```json
{ 
  "rewrites": [
    {
      "source": "/:path*",
      "has": [
        {
          "type": "host",
          "value": "en.website.com"
        }
      ],
      "destination": "/en/:path*"
    },
    {
      "source": "/",
      "has": [
        {
          "type": "host",
          "value": "en.website.com"
        }
      ],
      "destination": "/en"
    }
  ]
}
```

If that doesn't work for you, [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware#) is another option for rewrites and redirects that gives you more flexibility and control.

**Reply by mroddev** [reply]
> I want to make sure we're talking about the same thing. [Redirects](https://vercel.com/docs/concepts/projects/project-configuration#redirects) send users to a new URL which is displayed as the page address. [Rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites) display content from a different URL but do not change the original address that was accessed.
> 
> I don't know your site setup, so please forgive me if these questions seem silly. Should the content displayed at `website.com/en` be different from what is being shown at `en.website.com`? Do you get the result you expected if you change `rewrites` to `redirects`?
> 
> I know that sometimes rewriting the root path can get a bit of tricky. Does it work if you add a rewrite for the root path specifically? Then the vercel.json file would look like...
> 
> ```json
> { 
>   "rewrites": [
>     {
>       "source": "/:path*",
>       "has": [
>         {
>           "type": "host",
>           "value": "en.website.com"
>         }
>       ],
>       "destination": "/en/:path*"
>     },
>     {
>       "source": "/",
>       "has": [
>         {
>           "type": "host",
>           "value": "en.website.com"
>         }
>       ],
>       "destination": "/en"
>     }
>   ]
> }
> ```
> 
> If that doesn't work for you, [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware#) is another option for rewrites and redirects that gives you more flexibility and control.

After some digging I found what my problem was. Indeed I was looking for Redirects and not Rewrites here.
What I ended up using was pretty similar to what you just provided, instead I had to put the root redirect first for it not to be overwritten by the "/:path*" redirect and everything worked fine.
